community/kernel/src/main/java/org/neo4j/kernel/impl/store/id/IdGeneratorImpl.java

```diff
@@ -155,20 +156,11 @@ public synchronized long nextId()
             // special values, f.ex. the end of a relationships/property chain.
             id = highId.incrementAndGet();
         }
-        assertIdWithinCapacity( id );
+        IdValidator.assertValidId( id, max );
```

Comment:
I'd like it if IdValidator also had `isValidId` method which would do the above check... because now both IdGeneratorImpl and IdValidator knows about this exceptional ID and the double check which that brings looks slightly odd.